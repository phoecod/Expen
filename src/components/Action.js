import React from 'react';
import ReactModal from 'react-modal';

ReactModal.setAppElement('#app');

const action = class Action extends React.Component {
	
	constructor (props) {
		super(props);
		this.state = {
			selectedOption: undefined,
			showModal: false
		}
	}

	chooseOption = () => {
		const option = this.props.options[Math.floor(Math.random()*this.props.options.length)];
		if (!!option) {
			this.setState(() => {
				return {
					selectedOption: option,
					showModal: true
				}
			}); 
		}
	}

	handleCloseModal  = () => {
		this.setState({ showModal: false});
	}

	render = () => {
		const optionsLength = this.props.options.length > 0;
		return (
			<div>
				<button className="big-button" disabled={!optionsLength} onClick={this.chooseOption}>what should I do?</button>
				<ReactModal
					className = "modal"
					isOpen={this.state.showModal}
					onRequestClose={this.handleCloseModal}
					contentLabel="Selected Option"
					className="modal"
					closeTimeoutMS={500}
				>
					<h1 className=".modal-title">Selected Option</h1>
					<h2 className="modal-body">{this.state.selectedOption}</h2>
					<button className="button" onClick={this.handleCloseModal}>Back</button>
				</ReactModal>
			</div>
		);
	}
}

export default action;
