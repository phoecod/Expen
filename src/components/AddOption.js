import React from 'react';
import { CSSTransitionGroup } from 'react-transition-group';

export default class AddOption extends React.Component {

  state = {
    error: undefined
  }
  
  addToOptions = (e) => {    
    e.preventDefault();
    const value = e.target.elements.option.value.trim();
    const errorMsg = this.props.handleAddOption(value);
    this.setState(() => {
      return {
        error: errorMsg
      }
    });
    if (!errorMsg) {
      e.target.elements.option.value = '';
    }
  }
  render () {
    return (
      <div className="">
        {this.state.error && <p className="add-option-error">{this.state.error}</p>}
        <form className="add-option" onSubmit={this.addToOptions}>
          <input className="add-option-input" type="text" name="option"></input>
          <button className="button">Add Option</button>
        </form>
      </div>
      
    );
  }
}