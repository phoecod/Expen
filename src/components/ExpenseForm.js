import React from 'react';
import moment from 'moment';
import {connect} from 'react-redux';
import  { SingleDatePicker } from 'react-dates';



export const ExpenseForm = class ExpenseForm extends React.Component {

	constructor (props) {
		super (props);
		this.state = {
			description: props.expense ? props.expense.description : '',
			amount: props.expense ? props.expense.amount.toString() : '',
			note: props.expense ? props.expense.note : '',
			createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
			calendarFocused: false,
			descError: undefined, 
			amountError: undefined
		}
	}

	handleAddFormExpense = (e) => {
		e.preventDefault();
		const expense = {
			description: this.state.description,
			note: this.state.note,
			amount: parseFloat(this.state.amount, 10),
			createdAt: this.state.createdAt.valueOf()
		}
		if (!this.state.amount) {
			this.setState( () => ({amountError: 'Amount cannot be empty and needs to be greater than 0 up to 2 decimal places'}));
			this.setState( () => ({amount: ''}))
		}

		if (this.state.description.length == 0) {
			this.setState(() => ({descError: 'Description cannot be empty'}));
			this.setState(() => ({description: ''}));
		}

		if (!this.state.descError || !this.state.amountError) {
			this.props.onSubmit(expense);
		}
	}

	handleDescription = (e) => {
		let description = e.target.value.trim();
		if (description.length > 0 || !description ) {
			let descError = undefined;
			this.setState(() => ({description, descError}));
		}
	}

	handleAmount = (e) => {
		let amount = e.target.value.trim();
		if (amount.match(/^\d{1,}(\.\d{0,2})?$/) || !amount) {
			let amountError = undefined;
			this.setState(() => ({amount, amountError}));
		}
	}

	handleNote = (e) => {
		let note = e.target.value.trim();
		this.setState(() => ({note}));
	}

	onDateChange = (createdAt) => {
		this.setState(() => ( {createdAt} ));
	}

	onFocusChange = ({focused}) => {
		this.setState(() => ( {calendarFocused: focused} ));
	}
	
	formPh = {
		description: "Description",
		amount: "Amount",
		note: "Note"
	}

	render () {
		return (
			<div>
				<div className="page-header">
					<div className="form-frame">
						<h3 className="header-title">{this.props.expense ? 'Edit Expense' : 'Add Expense'}</h3>
					</div>
				</div> 	
				<div className="form-container">
					<div className="form-frame">
						<form className="form" onSubmit={this.handleAddFormExpense}>
							{this.state.descError && <p className="form-error">{this.state.descError}</p>}
							<input className="form-input text-input" value={this.state.description} onChange={this.handleDescription}  placeholder={this.formPh.description} name={this.formPh.description}/><br/>
							{this.state.amountError && <p className="form-error">{this.state.amountError}</p>}
							<input className="form-input text-input" value={this.state.amount} onChange={this.handleAmount} 
								placeholder={this.formPh.amount} name={this.formPh.amount}/><br/>
							<SingleDatePicker
								className="form-input"
								date={this.state.createdAt}
								onDateChange={this.onDateChange}
								focused={this.state.calendarFocused}
								onFocusChange={this.onFocusChange}
								numberOfMonths={1}
								isOutsideRange={() => false}
							/>
							<textarea className="form-input textarea" value={this.state.note}  onChange={this.handleNote} type="text" placeholder={this.formPh.note} name={this.formPh.note}/><br/>
							<div>
							<button className="btn">{this.props.expense ? 'Edit Expense' : 'Add Expense'}</button>
							</div>
							
						</form>
					</div>
				</div>
			</div>
		);
	}
	
}

const connectedExpenseForm = connect ()(ExpenseForm);

export default connectedExpenseForm;