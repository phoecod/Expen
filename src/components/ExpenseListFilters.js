import React from 'react';
import { connect } from 'react-redux';
import {setTextFilter, setSortByFilter, setStartDateFilter, setEndDateFilter} from '../actions/filters';
import {DateRangePicker} from 'react-dates';

export const ExpenseListFilters = class ExpenseListFilters extends React.Component {

	constructor (props) {
		super(props);
		this.state = {
			calendarFocused: null
		}
	}

	onDatesChange = ({startDate, endDate}) => {
		this.props.setStartDateFilter(startDate);
		this.props.setEndDateFilter(endDate);
	}

	onFocusChange = (calendarFocused) => {
		this.setState(() => ({calendarFocused}));
	}

	onTextChange = (e) => {
		const text = e.target.value;
		this.props.setTextFilter(text) ;
	}

	onSelectChange = (e) =>{
		this.props.setSortFilter(e.target.value);
	}

	render () {
		return (
			<div className="list-container">
				<div className="frame">
					<input id="search-expenses" placeholder="Search expenses" className="filter text-input" onChange={(e) => this.onTextChange(e)} type="text" value={this.props.filters.text} />
					<select className="filter select" onChange={(e) => this.onSelectChange(e)}>
						<option value="amount">Amount</option>
						<option value="date">Date</option>
					</select>
					<div className="filter">
						<DateRangePicker
						className="date-picker" 
						startDate={this.props.filters.startDate}
						endDate={this.props.filters.endDate}
						onDatesChange={this.onDatesChange}
						focusedInput={this.state.calendarFocused}
						onFocusChange={this.onFocusChange}
						numberOfMonths={2}
						isOutsideRange={() => false}
						showClearDates={true}
						/>
					</div>
				</div>
			</div>
		);
	}
	
}

const stateToProps = (state) => {
	return {
		filters: state.filters
	}
}

const mapDispatchtoProps = (dispatch) => ({
	setStartDateFilter: (startDate) => dispatch(setStartDateFilter(startDate)),
	setEndDateFilter: (endDate) => dispatch(setEndDateFilter(endDate)),
	setSortFilter: (e) => dispatch(setSortByFilter(e)),
	setTextFilter: (text) => dispatch(setTextFilter(text))
});

export default connect(stateToProps,mapDispatchtoProps)(ExpenseListFilters);