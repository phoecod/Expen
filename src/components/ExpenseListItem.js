import React from 'react';
import {startRemoveExpense} from '../actions/expenses';
import { connect } from 'react-redux';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
import {FaEdit, FaTrash} from 'react-icons/fa';




export class ExpenseListItem extends React.Component {

	onClick = () => {
		const itemId = this.props.id;
		this.props.startRemoveExpense(itemId);
	}

	render () {
		return (
			<div className="item-list">
				<div className="item-left">
					<h3 className="item-description"> {this.props.description}</h3> 
					{this.props.createdAt && <span className="item-date"> {moment(this.props.createdAt).format("MMM Do YYYY")}</span>}
					{this.props.note && <div>{this.props.note}</div>}
				</div>		
				<div className="item-right">
					<span className="item-amount">{numeral(this.props.amount).format('$0, 0.00')}</span> 
					<div className="item-btns">
						<button className="sm-btn delete" onClick={ this.onClick }>
							<FaTrash />
						</button>
						<Link className="sm-btn edit" to={`/edit/${this.props.id}`}>
							<FaEdit  />
						</Link>
					</div>
				</div>
			</div>
		);
	}
	
}

const mapDispatchToProps = (dispatch) => {
	return {
		startRemoveExpense: (id) =>  dispatch(startRemoveExpense({id}))
	}
}

export default connect (undefined, mapDispatchToProps)(ExpenseListItem);