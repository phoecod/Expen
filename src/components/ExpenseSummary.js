import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expensesTotal';
import {NavLink} from 'react-router-dom';
import numeral from 'numeral';

export const ExpenseSummary = ({expensesCount, expensesTotal}) => {

	const expenseWord = expensesCount > 1 ? 'expenses' : 'expense';
	const totalExpensesFormat = numeral(expensesTotal).format('$0.00');
	return (
		<div className="page-header">
			<div className="frame">
				{
					<h3 className="header-title"> 
						Viewing <span className="variable-text">{expensesCount}</span> {expenseWord} totalling <span className="variable-text">{totalExpensesFormat}</span>
					</h3>
					
				}
				<NavLink to="/create" className="summary-btn btn">Add Expense</NavLink>
			</div>
		</div>
	);
}

const mapStateToProps = (state) => {
	const visibleExpenses = selectExpenses(state.expenses, state.filters);
	return {
		expensesTotal: selectExpensesTotal(visibleExpenses),
		expensesCount: visibleExpenses.length
	}
}

export default connect(mapStateToProps)(ExpenseSummary);
