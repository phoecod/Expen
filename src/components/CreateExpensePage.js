import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {addExpense} from '../actions/expenses';

export class CreateExpensePage extends React.Component {

	onSubmit = (expense) => {
		this.props.addExpense(expense);
		this.props.history.push('/');
	}
	render() {
		return (
			<div>
				<h3>Add Expense</h3>
				<ExpenseForm onSubmit={this.onSubmit} />
			</div>
		)
	}
}
const mapDispatchtoProps = (dispatch) => ({
	addExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchtoProps)(CreateExpensePage);