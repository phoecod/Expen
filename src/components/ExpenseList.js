import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => {

	return (
		<div className="table-container">
			<div className="frame">
				<div className="table-header">
					<div className="show-for-mobile">Expense</div>
					<div className="show-for-desktop">Expense</div>
					<div className="show-for-desktop">Amount</div>
				</div>
				{
					props.expenses.length === 0 ? (
						<p> no expenses </p>
					) : (
					props.expenses.map((expense, index) => {
						return <ExpenseListItem key={index} {...expense}/>
					})
					)
				}
			</div>
		</div>
	);
}
const connectedExpenseList = connect ((state) => {
	return {
		expenses: selectExpenses(state.expenses, state.filters)
	}
})(ExpenseList);

export default connectedExpenseList;