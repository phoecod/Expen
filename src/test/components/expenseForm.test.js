import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseForm} from '../../components/ExpenseForm';
import expenseData from '../fixtures/expenseData';
import moment from 'moment';

test('should render expense form correctly', () => {
	const wrapper = shallow(<ExpenseForm />);
	expect(wrapper).toMatchSnapshot();
});

test('should render expense form with expense form', () => {
	const wrapper = shallow(<ExpenseForm  expense={expenseData[0]}/>);
	expect(wrapper).toMatchSnapshot();
});

test('should render expense form with no description', () => {
	const wrapper = shallow(<ExpenseForm />);
	wrapper.find('form').simulate('submit', {
		preventDefault: () => {}
	});
	expect(wrapper.state('description').length).toEqual(0);
});

test('should set description on input change', () => {
	const wrapper = shallow(<ExpenseForm/>);
	const value = 'desc';
	wrapper.find('input').at(0).simulate('change', {
		target: {value}
	});
	expect(wrapper.state('description')).toBe(value);
});

test('should set note on input change', () => {
	const wrapper = shallow(<ExpenseForm/>);
	const value = 'test';
	wrapper.find('textarea').simulate('change', {
		target: {value}
	});
	expect(wrapper.state('note')).toBe(value);
});

test('should set amount on input change', () => {
	const wrapper = shallow(<ExpenseForm/>);
	const value = '13.25';
	
	wrapper.find('input').at(1).simulate('change', {
		target: {value}
	});
	expect(wrapper.state('amount')).toBe(value);
});

test('should not set amount on input change as format is wrong', () => {
	const wrapper = shallow(<ExpenseForm/>);
	const value = '13.2564';
	
	wrapper.find('input').at(1).simulate('change', {
		target: {value}
	});
	expect(wrapper.state('amount')).toBe("");
});


test('should call onSubmit prop for valid form submission', () => {
	const onSubmitSpy = jest.fn();
	const wrapper = shallow(<ExpenseForm expense={expenseData[0]} onSubmit={onSubmitSpy}/>);
	wrapper.find('form').simulate('submit', {
		preventDefault: () => {}
	});
	expect(wrapper.state('amountError')).toBe(undefined);
	expect(onSubmitSpy).toHaveBeenLastCalledWith({
		description: expenseData[0].description,
		amount: expenseData[0].amount,
		note: expenseData[0].note,
		createdAt: expenseData[0].createdAt
	});
});

test('should call onDateChange prop', () => {
	const now = moment();
	const wrapper = shallow(<ExpenseForm />);
	wrapper.find('SingleDatePicker').prop('onDateChange')(now);
	expect(wrapper.state('createdAt')).toEqual(now);
});

test('should call onFocusChange prop', () => {
	const wrapper = shallow(<ExpenseForm />);
	wrapper.find('SingleDatePicker').prop('onFocusChange')({focused: true});
	expect(wrapper.state('calendarFocused')).toBe(true);
});