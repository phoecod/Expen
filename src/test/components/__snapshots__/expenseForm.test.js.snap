// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense form correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="form-frame"
    >
      <h3
        className="header-title"
      >
        Add Expense
      </h3>
    </div>
  </div>
  <div
    className="form-container"
  >
    <div
      className="form-frame"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          className="form-input text-input"
          name="Description"
          onChange={[Function]}
          placeholder="Description"
          value=""
        />
        <br />
        <input
          className="form-input text-input"
          name="Amount"
          onChange={[Function]}
          placeholder="Amount"
          value=""
        />
        <br />
        <SingleDatePicker
          anchorDirection="left"
          className="form-input"
          customCloseIcon={null}
          customInputIcon={null}
          date={"1970-01-01T00:00:00.000Z"}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          id="date"
          initialVisibleMonth={null}
          inputIconPosition="before"
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          numberOfMonths={1}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          openDirection="down"
          orientation="horizontal"
          phrases={
            Object {
              "calendarLabel": "Calendar",
              "chooseAvailableDate": [Function],
              "clearDate": "Clear Date",
              "closeDatePicker": "Close",
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month",
              "jumpToPrevMonth": "Move backward to switch to the previous month",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
              "moveFocusByOneMonth": "Switch months",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
              "openThisPanel": "Open this panel",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field",
              "selectFocusedDate": "Select the date in focus",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          placeholder="Date"
          readOnly={false}
          renderCalendarInfo={null}
          renderDay={null}
          renderMonth={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={false}
          weekDayFormat="dd"
          withFullScreenPortal={false}
          withPortal={false}
        />
        <textarea
          className="form-input textarea"
          name="Note"
          onChange={[Function]}
          placeholder="Note"
          type="text"
          value=""
        />
        <br />
        <div>
          <button
            className="btn"
          >
            Add Expense
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`should render expense form with expense form 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="form-frame"
    >
      <h3
        className="header-title"
      >
        Edit Expense
      </h3>
    </div>
  </div>
  <div
    className="form-container"
  >
    <div
      className="form-frame"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          className="form-input text-input"
          name="Description"
          onChange={[Function]}
          placeholder="Description"
          value="Gum"
        />
        <br />
        <input
          className="form-input text-input"
          name="Amount"
          onChange={[Function]}
          placeholder="Amount"
          value="195"
        />
        <br />
        <SingleDatePicker
          anchorDirection="left"
          className="form-input"
          customCloseIcon={null}
          customInputIcon={null}
          date={"1970-01-01T00:00:00.000Z"}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          id="date"
          initialVisibleMonth={null}
          inputIconPosition="before"
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          numberOfMonths={1}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          openDirection="down"
          orientation="horizontal"
          phrases={
            Object {
              "calendarLabel": "Calendar",
              "chooseAvailableDate": [Function],
              "clearDate": "Clear Date",
              "closeDatePicker": "Close",
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month",
              "jumpToPrevMonth": "Move backward to switch to the previous month",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
              "moveFocusByOneMonth": "Switch months",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
              "openThisPanel": "Open this panel",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field",
              "selectFocusedDate": "Select the date in focus",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          placeholder="Date"
          readOnly={false}
          renderCalendarInfo={null}
          renderDay={null}
          renderMonth={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={false}
          weekDayFormat="dd"
          withFullScreenPortal={false}
          withPortal={false}
        />
        <textarea
          className="form-input textarea"
          name="Note"
          onChange={[Function]}
          placeholder="Note"
          type="text"
          value=""
        />
        <br />
        <div>
          <button
            className="btn"
          >
            Edit Expense
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
